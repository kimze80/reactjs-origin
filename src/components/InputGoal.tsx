import * as React from 'react';
import { InputGroup, FormControl } from 'react-bootstrap';

type InputGoalProps = {
  inputValue: any;
  inputChange: any;
}

const InputGoal: React.FC<InputGoalProps> = ({inputValue, inputChange}) => {
   
    return (
      <div>
        <InputGroup className="input-goal">
          <InputGroup.Prepend>
            <InputGroup.Text id="currency-icon">$</InputGroup.Text>
          </InputGroup.Prepend>
          <FormControl
            placeholder="20,000"
            aria-label="Username"
            aria-describedby="currency-icon"
            onChange={inputChange}
            value={inputValue}
          />
        </InputGroup>
      </div>
    )
}
export default InputGoal;
