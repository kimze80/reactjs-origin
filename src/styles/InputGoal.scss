
.input-goal {
  // border: #eff7ff;
  #currency-icon {
    font-size: 22px;
    background: #eff7ff;
    color: #657786;
    border: #e1e8ed solid 0.5px;
    padding: 10px 20px;
  }

  input {
    border: #e1e8ed solid 0.5px;
    height: 56px;
    font-size: 20px;
  }
}

@media (max-width: 576px) {
  .input-goal {
    margin-bottom: 15px;
  }
}