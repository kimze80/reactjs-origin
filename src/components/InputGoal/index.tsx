import React, {useState, useCallback} from 'react';
import { InputGroup, FormControl } from 'react-bootstrap';
import { Container } from './styles';
import numeral from 'numeral';

interface InputGoalProps {
  goalValueProp: Function;
}

const InputGoal: React.FC<InputGoalProps> = ({goalValueProp}) => {
  const [value, setValue] = useState('');

  const onchange = useCallback((e: any) => {
    let val = e.target.value

    // it will format in currency number Ex: 1,000
    let withComma = numeral(val).format('0,0')
    setValue(withComma)

    // it will get the value, take the comma off and format in number
    let num = Number(val.replace(/\,/g,''))

    // it will send the number to parent component
    goalValueProp(num, withComma)
  }, [])

  return (
    <Container>
      <InputGroup className="input-goal">
        <InputGroup.Prepend>
          <InputGroup.Text id="currency-icon">$</InputGroup.Text>
        </InputGroup.Prepend>
        <FormControl
          placeholder="20,000"
          aria-label="currency"
          aria-describedby="currency-icon"
          onChange={onchange}
          value={value ? value : ''}
        />
      </InputGroup>
    </Container>
  )
}
export default InputGoal;
