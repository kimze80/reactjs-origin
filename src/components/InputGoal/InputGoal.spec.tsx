import * as React from 'react';
import { shallow } from 'enzyme';
import InputGoal from './index';

describe('Testing InputGoal', () => {
  it('Should render InputGoal correctly', () => {
    const func = () => {return '3,000'}
    const wrapper = shallow(
      <InputGoal goalValueProp={func}/>
    );

    expect(wrapper).toMatchSnapshot();
  })
})