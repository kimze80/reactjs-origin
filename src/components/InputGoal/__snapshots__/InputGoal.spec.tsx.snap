// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing InputGoal Should render InputGoal correctly 1`] = `ShallowWrapper {}`;
